### DISCUSSION TITLE
[@types/react-native] Regression in TouchableOpacityProps since version 0.65.7

### DISCUSSION THREAD

**Discussion by gvillenave** [DefinitelyTyped/DefinitelyTyped#56737] [Issues with a @types package]
Since `@types/react-native >= 0.65.7`, TouchableOpacityProps now extends TVProps, which causes the following TypeScript error when consuming TouchableOpacity, due to the missing `hasTVPreferredFocus` and `tvParallaxProperties` which are specific to the Apple TV platform:

```
error TS2739: Type '{ children: Element[]; bottomDivider: boolean; disabled: boolean | undefined; onPress: (() => void)
 | undefined; }' is missing the following properties from type 'Pick<TouchableHighlightProps & { containerStyle?: 
StyleProp<ViewStyle>; disabledStyle?: StyleProp<ViewStyle>; ... 6 more ...; children?: any; } & Partial<...>, "children"
 | ... 45 more ... | "bottomDivider">': hasTVPreferredFocus, tvParallaxProperties
```

This regression is likely due to https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56374

Related issues:
https://github.com/react-native-elements/react-native-elements/issues/3227
https://github.com/callstack/react-native-paper/issues/2929

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
`TouchableOpacity` does not implement `tvParallaxProperties` on 0.65 according to react-native's own flow types:
- https://github.com/facebook/react-native/blob/v0.65.0/Libraries/Components/Touchable/TouchableOpacity.js#L23-L40
- https://github.com/facebook/react-native/blob/v0.65.0/Libraries/Components/Touchable/TouchableWithoutFeedback.js#L32-L65

`tvParallaxProperties` is falsely listed on the docs. That change was intentional:

>  Some docs which mistakenly state a prop that don't actually do anything. I've left this in for now but a note will have to be made to change this one the confusion on docs is cleared up TouchableOpacity has tvParallaxProperties, which doesn't do anything in src

-- https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56374

`hasTVPreferredFocus` is missing from `TVProps` though.

**Comment by cbotsikas** [comment]
I also have the same issue with `hasTVPreferredFocus`, `tvParallaxProperties` and `autoCompleteType`

lib | version
-- | --
react-native-elements | 3.4.2
@types/react-native | 0.66.1
react-native | 0.66.1

On a `ListItem` from `react-native-elements`:
```
error TS2739: Type '{ children: Element[]; containerStyle: { alignContent: "space-around"; alignItems: "stretch"; backgroundColor: string; borderColor: string; borderRadius: number; borderWidth: number; flex: number; flexDirection: "column"; paddingHorizontal: number; paddingVertical: number; }; }' is missing the following properties from type 'Pick<TouchableHighlightProps & { containerStyle?: StyleProp<ViewStyle>; disabledStyle?: StyleProp<ViewStyle>; ... 6 more ...; children?: any; } & Partial<...>, "testID" | ... 45 more ... | "bottomDivider">': hasTVPreferredFocus, tvParallaxProperties
```

On an `Input` from `react-native-elements`
```
error TS2741: Property 'autoCompleteType' is missing in type '{ value: string | undefined; placeholderTextColor: string; inputStyle: { fontWeight: "bold"; color: string; fontSize: number; paddingLeft: number; }; placeholder: string; leftIcon: Element; containerStyle: { ...; }; inputContainerStyle: { ...; }; onChangeText: (inputValue: string) => void; }' but required in type 'Pick<TextInputProps & RefAttributes<TextInput> & { containerStyle?: StyleProp<ViewStyle>; disabled?: boolean | undefined; ... 14 more ...; renderErrorMessage?: boolean | undefined; } & Partial<...>, "testID" | ... 125 more ... | "renderErrorMessage">'.
```

Related issue: https://github.com/react-native-elements/react-native-elements/issues/3231
